psgo.go

```diff
@@ -334,19 +334,64 @@ func JoinNamespaceAndProcessInfo(pid string, descriptors []string) ([][]string,
 	return data, dataErr
 }
 
+// JoinNamespaceAndProcessInfoByPids has similar semantics to
+// JoinNamespaceAndProcessInfo and avoids duplicate entries by joining a giving
+// PID namepsace only once.
+func JoinNamespaceAndProcessInfoByPids(pids []string, descriptors []string) ([][]string, error) {
+	pidList := []string{}
```

Comment:
nit: why have both pidList and nsMap? if we only add the pid to pidList when it's added to nsMap, couldn't we just iterate through the values? 
If it's to keep track of which is the first, couldn't a bool be used instead of an entire slice?

Reply:
The big problem with maps in golang is that they are nondeterministic. Without the [determinisitic] list, we'd potentially get different results for the same calls on the same data. The map, however, is much faster when checking for values, so I ended up with the two. The list to preserve the order, the map for fast lookups.

Reply:
Okay makes sense to me. /me marks comment as resolved

Reply:
> If it's to keep track of which is the first, couldn't a bool be used instead of an entire slice?

That's a nice idea. Then we don't need to store strings and just have a `map[string]bool`. I'll update the PR tomorrow with (potentially) some other comments from @rhatdan.